### DISCUSSION TITLE
Runtime error after installing 2sxc 18.04.00 in Oqtane version 6

### DISCUSSION THREAD

**Discussion by patrickjlee** [2sic/2sxc#3514] [General] [answered]
hi - I have watched the various videos today about 2sxc and am very excited to try it out. 
But I've now tried twice to get it running with my (so far local, i.e. running in debug) Oqtane 6 (.NET 9) installation, but each time I get run time errors.

E.g. I've just done a clean install of Oqtane 9, and then went to Module Management and installed the 2sxc 18.04.00 module from there.  I restarted Oqtane 9, and then created two new pages.  For the first of these I added a Content module, for the second I added an App module.
But when I go to either age I get a run time error:

Can't find uiRoot - something went wrong, pls contact 2sxc.org

When I click on F5 to continue I then get a second error (perhaps not surprisingly given the first one):

Can't load pageid, moduleid, etc. and $ is not available. 
You must either include jQuery on the page or inject the jsApi parameters to prevent auto-detection.

Any idea how I can solve this please?

**Comment by iJungleboy** [comment]
@patrickjlee glad that you want to try it out ;)

Are you running in Visual Studio or on an IIS?

If Visual Studio, did you disable hot reloading as is explained in the warnings & docs?

**Reply by patrickjlee** [reply]
Thanks, I was running in Visual Studio locally, and then later in Azure as an app service.
I didn't see the warning to disable hot reloading / where is that in the docs please?

**Reply by iJungleboy** [reply]
when running in Visual studio, pls check https://docs.2sxc.org/abyss/platforms/oqtane/install-issue-hot-reload.html

Do you also have the problem in Azure running "normally"?

**Reply by patrickjlee** [reply]
Thanks! That (adding "hotReloadEnabled" : false to  launchSettings.json) enabled me to get over that hurdle, and I was able to create a page and add a 2sxc Content module to that page. I then chose the recommended option of "install recommended content package" and chose to add the first option, basic content and then I chose image and text, which appeared.

I then clicked the pencil icon (as suggested) to edit the content, but I then got a Navigation error (which I understand is a current problem I understand with Blazor), I clicked F5 and then tried to restart the site. But now the solution fails to build, I get build errors: 
"The type or namespace name "ToSic" could not be found and similar - even though I had restarted Oqtane after installing the 2sxc module):

![image](https://github.com/user-attachments/assets/5c109397-3b4a-4ff7-bb95-07aca071dbfc)


Sorry if I am being silly here, what should I have done to avoid this problem? 

PS I haven't yet have time to try running again with the hotReloadEnabled set to false setting in Azure.  The above is just running locally in debug mode.

**Reply by patrickjlee** [reply]
I've tried adding 2sxc several times now in different Oqtane version 6 fresh installs locally, and each time (with the hotReloadEnabled set to false) I can only get as far as installing the module, but after adding either Content or an App to a page, when I rebuild the site it has build errors (in fact it had 1159 such errors after installing the 23 popular apps).

Unfortunately this then makes Oqtane unusable

**Comment by tvatavuk** [comment, marked as answer]
I'm repeating this comment from #3515 for others.

A month ago, I created a YouTube video that demonstrates all the steps to install and run the Oqtane source version with the 2sxc module, including the 2sxc Content and App templates, all using Visual Studio.

[Install Oqtane and 2sxc in Microsoft Visual Studio](https://www.youtube.com/watch?v=dlDnD9QfVHc)

Please make sure to exclude the following folders from the Oqtane.Server project in Visual Studio before building and running Oqtane:

- `2sxc`
- `App_Data`
- `Content`
- `wwwroot\Modules\ToSic.Sxc.Oqtane`

These steps are demonstrated in the video around the 6:40 mark: [Exclude Folders in Visual Studio](https://youtu.be/dlDnD9QfVHc?t=400). You can also see an example of the build error that occurs if you skip these steps.

Alternatively, you can manually add the following to the `Oqtane.Server.csproj` file to exclude these directories from being compiled:

```csproj
<ItemGroup>
  <!-- Exclude all files in directory and its subdirectories from being compiled. -->
  <Compile Remove="2sxc\**" />
  <Compile Remove="App_Data\**" />
  <Compile Remove="Content\**" />
  <Compile Remove="wwwroot\Modules\ToSic.Sxc.Oqtane\**" />
  <!-- Exclude all content files (like .html, .css, .js) from being included in the build output. -->
  <Content Remove="2sxc\**" />
  <Content Remove="App_Data\**" />
  <Content Remove="Content\**" />
  <Content Remove="wwwroot\Modules\ToSic.Sxc.Oqtane\**" />
  <!-- Exclude all files from being embedded as resources in the assembly. -->
  <EmbeddedResource Remove="2sxc\**" />
  <EmbeddedResource Remove="App_Data\**" />
  <EmbeddedResource Remove="Content\**" />
  <EmbeddedResource Remove="wwwroot\Modules\ToSic.Sxc.Oqtane\**" />
  <!-- Exclude files that are not otherwise included in any other item group (e.g., documentation files). -->
  <None Remove="2sxc\**" />
  <None Remove="App_Data\**" />
  <None Remove="Content\**" />
  <None Remove="wwwroot\Modules\ToSic.Sxc.Oqtane\**" />
  <!-- Exclude all files from being watched by the dotnet watch tool. -->
  <Watch Remove="2sxc\**" />
  <Watch Remove="App_Data\**" />
  <Watch Remove="Content\**" />
  <Watch Remove="wwwroot\Modules\ToSic.Sxc.Oqtane\**" />
  <!-- Exclude all TypeScript files in this directory and subdirectories -->
  <TypeScriptCompile Remove="2sxc\**" />
  <TypeScriptCompile Remove="App_Data\**" />
  <TypeScriptCompile Remove="Content\**" />
  <TypeScriptCompile Remove="wwwroot\Modules\ToSic.Sxc.Oqtane\**" />
</ItemGroup>
```

This configuration explicitly defines exclusions for the `2sxc`, `App_Data`, `Content`, and `wwwroot\Modules\ToSic.Sxc.Oqtane` directories.

These folders contain additional assets that are part of the 2sxc apps and templates. While they are used at runtime, they are not intended to be included as source files in the Oqtane.Server project and should not be compiled as part of it. Visual Studio will automatically include them after 2sxc Content or App templates are downloaded, installed, or created. Including them may cause issues during the build process. Therefore, please exclude them before compiling and running the Oqtane development environment.

**Comment by patrickjlee** [comment]
Thanks, that fixed it.  I suggest that the need to add those exclusions to the csproj file be added to the installation instructions to save anyone else from wasting time encountering this issue.
